File: api/src/main/java/com/sybd/znld/web/onenet/dto/GetLastDataStreamsResult.java
package com.sybd.znld.web.onenet.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.sybd.znld.web.onenet.dto.BaseResult;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter @Setter
public class GetLastDataStreamsResult extends BaseResult {
    public Data data;
    public boolean isOk(){
        return super.errno == 0 && !data.getDevices().isEmpty();
    }

    @Getter @Setter
    public static class DataStream{
        public String at;
        public String id;
        public Object value;
    }
    @Getter @Setter
    public static class Device{
        public String title;
        public String id;
        @JsonProperty("datastreams")
        public List<DataStream> dataStreams;
    }
    @Getter @Setter
    public static class Data{
        public List<Device> devices;
    }
}

File: znld-web/src/main/java/com/sybd/znld/onenet/dto/GetLastDataStreamsResult.kt
package com.sybd.znld.onenet.dto;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.jetbrains.annotations.NotNull;

class GetLastDataStreamsResult(val data: Data, errno: Int, error: String): BaseResult(errno, error) {

    fun isOk(): Boolean{
        return errno == 0 && data.devices.isNotEmpty();
    }

    inner class Data(val devices: Array<Device>)
    inner class Device(val title: String, val id: String, val datastreams: Array<DataStream>)
    inner class DataStream(val at: String, val id: String, var value:Any)
}
